core/jvm/src/main/scala/zio/metrics/jvm/BufferPools.scala

```diff
@@ -0,0 +1,45 @@
+package zio.metrics.jvm
+
+import zio._
+
+import java.lang.management.{BufferPoolMXBean, ManagementFactory}
+
+object BufferPools extends JvmMetrics {
+
+  /** Used bytes of a given JVM buffer pool. */
+  private def bufferPoolUsedBytes(pool: String): ZIOMetric.Gauge[Long] =
+    ZIOMetric.setGaugeWith("jvm_buffer_pool_used_bytes", MetricLabel("pool", pool))(_.toDouble)
+
+  /** Bytes capacity of a given JVM buffer pool. */
+  private def bufferPoolCapacityBytes(pool: String): ZIOMetric.Gauge[Long] =
+    ZIOMetric.setGaugeWith("jvm_buffer_pool_capacity_bytes", MetricLabel("pool", pool))(_.toDouble)
+
+  /** Used buffers of a given JVM buffer pool. */
+  private def bufferPoolUsedBuffers(pool: String): ZIOMetric.Gauge[Long] =
+    ZIOMetric.setGaugeWith("jvm_buffer_pool_used_buffers", MetricLabel("pool", pool))(_.toDouble)
+
+  private def reportBufferPoolMetrics(
+    bufferPoolMXBeans: List[BufferPoolMXBean]
+  ): ZIO[Any, Throwable, Unit] =
+    ZIO.foreachParDiscard(bufferPoolMXBeans) { bufferPoolMXBean =>
+      for {
+        name <- Task(bufferPoolMXBean.getName)
+        _    <- Task(bufferPoolMXBean.getMemoryUsed) @@ bufferPoolUsedBytes(name)
+        _    <- Task(bufferPoolMXBean.getTotalCapacity) @@ bufferPoolCapacityBytes(name)
+        _    <- Task(bufferPoolMXBean.getCount) @@ bufferPoolUsedBuffers(name)
+      } yield ()
+    }
+
+  val collectMetrics: ZManaged[Has[Clock], Throwable, Unit] =
```

Comment:
Perhaps this should be a layer so it is more idiomatic if users want to collect just one type of JVM metric versus all of them? You can even imagine each of these being its own `ZIOApp` and then the `ZIOApp` that starts all of them is just the composition of those applications.